code/modules/food&drinks/food/snacks.dm

```diff
@@ -151,7 +152,10 @@
 			return 1
 
 //Called when you finish tablecrafting a snack.
-/obj/item/weapon/reagent_containers/food/snacks/CheckParts()
+/obj/item/weapon/reagent_containers/food/snacks/CheckParts(list/content)
+	..()
+	create_reagents(volume)
```

Comment:
You need to put it before ..() Since the reagent transfer is done in reagent_containers/CheckParts()